### DISCUSSION TITLE
CoAP Gateway -- Message retained flag?

### DISCUSSION THREAD

**Discussion by pearson** [emqx/emqx#5942] [Q&A] [answered]
Using the CoAP Gateway, is it possible to flag a CoAP message that should be retained by the MQTT server?

**Comment by hjianbo** [comment, marked as answer]
This configuration was not available in previous versions. Recently in 5.0, we have refactored CoAP and this feature is planned to be included in it

**Reply by pearson** [reply]
That's excellent news! It will certainly simplify some scenarios that I'm looking at.

What is the timeline for the 5.0 release?

**Reply by hjianbo** [reply]
The first RC release will probably be around December, and the official version of 5.0 is expected to be in January next year
